src/test.rs

```diff
@@ -482,3 +484,162 @@ fn it_works_get_pact() {
 
     assert_eq!(contract_without, None);
 }
+
+#[test]
+fn wildcard_method() {
+    let method = Method::new("*").block_cooldown(123);
+    let methods = make_methods(&method);
+
+    let module = Module::new("module_test")
+        .block_cooldown(1)
+        .methods(methods);
+
+    let result = module.get_method("my_unregistered_method");
+    assert_ne!(result, None);
+}
+
+#[test]
+fn wildcard_method_validates() {
+    let method = Method::new("*").block_cooldown(123);
+    let methods = make_methods(&method);
+
+    let module = Module::new("module_test")
+        .block_cooldown(1)
+        .methods(methods);
+    let modules = make_modules(&module);
+
+    let cennznut = CENNZnutV0 { modules };
+    let args = [];
+
+    assert_eq!(
+        cennznut.validate(&module.name, "my_unregistered_method", &args),
+        Ok(())
+    );
+}
+
+#[test]
+fn wildcard_module() {
+    let method = Method::new("registered_method").block_cooldown(123);
+    let methods = make_methods(&method);
+
+    let module = Module::new("*").block_cooldown(1).methods(methods);
+    let modules = make_modules(&module);
+
+    let cennznut = CENNZnutV0 { modules };
+
+    let result = cennznut.get_module("my_unregistered_module");
+    assert_ne!(result, None);
+}
+
+#[test]
+fn wildcard_module_validates() {
+    let method = Method::new("registered_method").block_cooldown(123);
+    let methods = make_methods(&method);
+
+    let module = Module::new("*").block_cooldown(1).methods(methods);
+    let modules = make_modules(&module);
+
+    let cennznut = CENNZnutV0 { modules };
+    let args = [];
+
+    assert_eq!(
+        cennznut.validate("my_unregistered_module", "registered_method", &args),
+        Ok(())
+    );
+}
+
+#[test]
+fn wildcard_module_wildcard_method_validates() {
+    let method = Method::new("*").block_cooldown(123);
+    let methods = make_methods(&method);
+
+    let module = Module::new("*").block_cooldown(1).methods(methods);
+    let modules = make_modules(&module);
+
+    let cennznut = CENNZnutV0 { modules };
+    let args = [];
+
+    assert_eq!(
+        cennznut.validate("my_unregistered_module", "my_unregistered_method", &args),
+        Ok(())
+    );
+}
+
+#[test]
+fn unregistered_module_fails_validation() {
+    let method = Method::new("registered_method").block_cooldown(123);
+    let methods = make_methods(&method);
+
+    let module = Module::new("registered_module")
+        .block_cooldown(1)
+        .methods(methods);
+    let modules = make_modules(&module);
+
+    let cennznut = CENNZnutV0 { modules };
+    let args = [];
+
+    assert_ne!(
```

Comment:
Would it be better to assert the expected error condition instead of "anything but `Ok(())`"